triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/test/TriggerIt.scala

```diff
@@ -0,0 +1,247 @@
+      Some(
+        Commands(
+          party = party,
+          commands = commands,
+          ledgerId = client.ledgerId.unwrap,
+          applicationId = MockMessages.applicationId,
+          commandId = UUID.randomUUID.toString
+        )))
+    for {
+      response <- client.commandServiceClient.submitAndWaitForTransaction(request)
+    } yield response.getTransaction.events.head.getCreated.contractId
+  }
+
+  private def archive(
+      client: LedgerClient,
+      party: String,
+      templateId: LedgerApi.Identifier,
+      contractId: String): Future[Unit] = {
+    val commands = Seq(
+      Command().withExercise(
+        ExerciseCommand(
+          templateId = Some(templateId),
+          contractId = contractId,
+          choice = "Archive",
+          choiceArgument = Some(LedgerApi.Value().withRecord(LedgerApi.Record())))))
+    val request = SubmitAndWaitRequest(
+      Some(
+        Commands(
+          party = party,
+          commands = commands,
+          ledgerId = client.ledgerId.unwrap,
+          applicationId = MockMessages.applicationId,
+          commandId = UUID.randomUUID.toString
+        )))
+    for {
+      response <- client.commandServiceClient.submitAndWaitForTransaction(request)
+    } yield ()
+  }
+
+  private def queryACS(client: LedgerClient, party: String) = {
+    val filter = TransactionFilter(List((party, Filters.defaultInstance)).toMap)
+    val contractsF: Future[Seq[CreatedEvent]] = client.activeContractSetClient
+      .getActiveContracts(filter, verbose = true)
+      .runWith(Sink.seq)
+      .map(_.flatMap(x => x.activeContracts))
+    contractsF.map(contracts =>
+      contracts.map(created => (created.getTemplateId, created.getCreateArguments)).groupBy(_._1))
+  }
+
+  "Trigger" can {
+    "AcsTests" should {
+      val assetId = LedgerApi.Identifier(packageId, "ACS", "Asset")
+      val assetMirrorId = LedgerApi.Identifier(packageId, "ACS", "AssetMirror")
+      def asset(party: String): CreateCommand =
+        CreateCommand(
+          templateId = Some(assetId),
+          createArguments = Some(
+            LedgerApi.Record(fields =
+              Seq(LedgerApi.RecordField("issuer", Some(LedgerApi.Value().withParty(party)))))))
+
+      final case class AssetResult(
+          successfulCompletions: Long,
+          failedCompletions: Long,
+          activeAssets: Set[String])
+
+      def toResult(expr: SExpr): AssetResult = {
+        val fields = expr.asInstanceOf[SEValue].v.asInstanceOf[SRecord].values
+        AssetResult(
+          successfulCompletions = fields.get(1).asInstanceOf[SInt64].value,
+          failedCompletions = fields.get(2).asInstanceOf[SInt64].value,
+          activeAssets = fields
+            .get(0)
+            .asInstanceOf[SList]
+            .list
+            .map(x =>
+              x.asInstanceOf[SContractId].value.asInstanceOf[AbsoluteContractId].coid.toString)
+            .toSet
+        )
+      }
+
+      "1 create" in {
```

Comment:
There is a bit of boilerplate between the tests that we should probably factor out but I wanted to have a few explicit tests for the first version.